JoinColony/colonyCDapp
# [Dashboard] feat: implement new team reputation chart

## Description

Using `nivo` implement the new pie chart for displaying reputation for teams (not the contributors yet tho!).
[Figma link](https://www.figma.com/design/gac6xVgGNCAge5Bia933dp/Dashboard?node-id=4822-31570&t=iooTLbqkuaIq753b-4)
![image](https://github.com/user-attachments/assets/6b5e894f-4354-49f8-88a8-6a32e5af1ca8)

## Testing

1. Run your dev env and the `create-data` script. Go to the dashboard and observe the chart. Make sure it's according to the designs.
![image](https://github.com/user-attachments/assets/0d7b6a34-9156-45b2-ac39-c25958656125)
2. Make sure that on hover the tooltip shows up 
![image](https://github.com/user-attachments/assets/1c4327dc-02fd-4dc2-bcee-448ec12ab420)
3. This is it for the static one, we need to check out two more edge cases so go ahead and run `node ./scripts/create-colony-url.js` and create a new colony with a custom token.
4. Make sure that the empty state shows up immediately:
![image](https://github.com/user-attachments/assets/4fc0a51a-9152-4839-bd67-6f933aa11c43)

There should be a way of testing the state where there is only reputation in `Root` but for some reason my dev env is acting up and I can't really provide screenshots and I can't get reputation (I'll try again at a different time_, but the gist of it is:
1. Install the voting reputation extension, mint some tokens and send them to `leela`
2. Create any motion in `General` fully stake it, `npm run forward-time 1`, refresh the page, finalize the motion and after a refresh, this widget state should show up
![image](https://github.com/user-attachments/assets/59269578-a7ed-4585-a954-185ca0fd2ead)
3. Creating a new team and creating a motion inside it (to edit the team details for example), staking, forwarding the time, finalizing it, should then give some reputation in that team and the chart should change to not show the `General` team anymore


## Diffs

**New stuff** ✨

* ReputationChart now just renders a nivo pie chart (3 new partial components)
* Temporary custom tooltip to be swapped with the correct one

**Changes** 🏗

* The chart item is now typed
* ` getTeamReputationChartData` now returns cummulative data for all other teams as the last chart entry
* had to add 2 more colors to the `summaryLegendColor` `const` so we can use the colors in tailwind without constant className regeneration

**Deletions** ⚰️

* `DonutChart` component

Resolves #2857
